Modules/_zoneinfo.c

```diff
@@ -2243,72 +2246,123 @@ ymd_to_ord(int y, int m, int d)
+        PyObject *num = PyObject_CallMethod(dt, "year", NULL);
         if (num == NULL) {
-            return -1;
+            return NULL;
+        }
+        year = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (year == -1) {
+            return NULL;
         }
 
-        ord = PyLong_AsLong(num);
+        num = PyObject_GetAttrString(dt, "month");
+        if (num == NULL) {
+            return NULL;
+        }
+        month = PyLong_AsLong(num);
         Py_DECREF(num);
-        if (ord == -1 && PyErr_Occurred()) {
-            return -1;
+        if (month == -1) {
+            return NULL;
+        }
+
+        num = PyObject_GetAttrString(dt, "day");
+        if (num == NULL) {
+            return NULL;
+        }
+        day = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (day == -1) {
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "hour");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         hour = PyLong_AsLong(num);
         Py_DECREF(num);
         if (hour == -1) {
-            return -1;
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "minute");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         minute = PyLong_AsLong(num);
         Py_DECREF(num);
         if (minute == -1) {
-            return -1;
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "second");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         second = PyLong_AsLong(num);
         Py_DECREF(num);
         if (second == -1) {
-            return -1;
+            return NULL;
         }
+
+        num = PyObject_GetAttrString(dt, "fold");
+        if (num == NULL) {
+            return NULL;
+        }
+        fold = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (fold > 1) {
+            return NULL;
+        }
+    }
+
+    return (PyDateTime_DateTime *) PyDateTime_FromDateAndTimeAndFold(
+        year, month, day, hour, minute, second, 0, fold
+    );
+}
+
+/* Calculate the number of seconds since 1970-01-01 in local time.
+ *
+ * This gets a datetime in the same "units" as self->trans_list_wall so that we
+ * can easily determine which transitions a datetime falls between. See the
+ * comment above ts_to_local for more information.
+ * */
+static int
+get_local_timestamp(PyObject *dt, int64_t *local_ts)
+{
+    assert(local_ts != NULL);
+
+    PyDateTime_DateTime *dt_dt = as_datetime(dt);
+    if (dt_dt == NULL) {
+        return -1;
     }
 
+    int year = PyDateTime_GET_YEAR(dt_dt);
+    int month = PyDateTime_GET_MONTH(dt_dt);
+    int day = PyDateTime_GET_DAY(dt_dt);
+    int hour = PyDateTime_DATE_GET_HOUR(dt_dt);
+    int minute = PyDateTime_DATE_GET_MINUTE(dt_dt);
+    int second = PyDateTime_DATE_GET_SECOND(dt_dt);
+    int ord = ymd_to_ord(year, month, day);
+
```

Comment:
```suggestion
    Py_DECREF(dt_dt);
```
We allocated a new thing, so we're on the hook to free it.

Reply:
Updated, thanks